File: 3871-count-commas-in-range-ii/3871-count-commas-in-range-ii.py
class Solution:
    def countCommas(self, n: int) -> int:
        answer = 0;

        power =1000;
        commas = 1;

        while (power <= n):
            end = min (n,power * 1000 -1)

            count = end -power +1

            answer += count *commas 

            power *= 1000
            commas  += 1

        return answer 

File: 3871-count-commas-in-range-ii/3871-count-commas-in-range-ii.java
class Solution {
    public long countCommas(long n) {
        long answer = 0;

        long power = 1000;
        long commas = 1;

        while (power <= n){
            long end = Math.min(n, power * 1000 -1);

            long count = end - power  +1;

            answer  += count * commas;

            power *= 1000;
            commas++;

        }
        return answer;
    }
}